docs/design/seccomp.md

```diff
@@ -0,0 +1,380 @@
+  // userland as the errno without executing the system call.
+  Errno SeccompAction = "Errno"
+  // The Trap action results in kernel attempting to notify a ptrace()-based tracer prior
+  // to executing the system call.  If there is no tracer present, ENOSYS is returned to
+  // userland and the system call is not executed.
+	Trace SeccompAction = "Trace"
+  // The Allow action results in the system call being executed.
+  Allow SeccompAction = "Allow"
+)
+
+// A comparison operator to be used when matching syscall arguments in Seccomp
+type SeccompOperator string
+
+const (
+	EqualTo              SeccompOperator = "EqualTo"
+	NotEqualTo           SeccompOperator = "NotEqualTo"
+	GreaterThan          SeccompOperator = "GreaterThan"
+	GreaterThanOrEqualTo SeccompOperator = "GreaterThanOrEqualTo"
+	LessThan             SeccompOperator = "LessThan"
+	LessThanOrEqualTo    SeccompOperator = "LessThanOrEqualTo"
+	MaskEqualTo          SeccompOperator = "MaskEqualTo"
+)
+
+// Represents a specific syscall argument in a seccomp rule.
+type SeccompArg struct {
+	Index    int             `json:"index"`
+	Value    int64           `json:"value"`
+	ValueTwo int64           `json:"valueTwo"`
+	Op       SeccompOperator `json:"op"`
+}
+```
+
+Along with this API change, there will be supporting code changes:
+
+1.  New REST API implementation
+2.  Client code
+2.  Validations/conversions
+3.  `kubectl` support (including describer)
+
+### Pod API changes
+
+The container `securityContext` must be changed to allow specifying a seccomp profile:
+
+```go
+// SecurityContext holds security configuration that will be applied to a container.
+// Some fields are present in both SecurityContext and PodSecurityContext.  When both
+// are set, the values in SecurityContext take precedence.
+type SecurityContext struct {
+  // other fields omitted
+  // SeccompProfileName is the name of the seccomp profile this pod should run under.
+  SeccompProfileName string `json:"seccompProfileName,omitempty"`
+}
+```
+
+The kubelet docker runtime implementation in `pkg/kubelet/dockertools` will
+require code changes to implement setting the seccomp profile when a container
+is run.  This API change should not affect the kubelet itself.
+
+The kubectl describer for pods should change to include the new field.
+
+### Pod Security Policy changes
+
+The `PodSecurityPolicy` type should be changed to control seccomp profiles:
+
+```go
+// PodSecurityPolicySpec defines the policy enforced.
+type PodSecurityPolicySpec struct {
+  // other fields omitted
+  Seccomp *SeccompStrategyOptions `json:"seccomp,omitempty"`
+}
+
+// Defines the seccomp strategy type and any options used to create the strategy.
+type SeccompStrategyOptions struct {
+	// Rule is the strategy that will dictate the allowable seccomp profile name values that may be set.
+	Rule SeccompStrategy `json:"rule"`
+	// Profiles are a list of profile names that may be used.
+	Profiles []string `json:"profiles,omitempty"`
```

Comment:
This would allow a single `PodSecurityPolicy` the ability to admit requests using a list of allowable profiles.  You can still only use one per container.